README.md

```diff
@@ -213,17 +213,17 @@ final class MyTool
      * @param int    $number The number of an object
      */
     public function __invoke(
-        #[ToolParameter(pattern: '/([a-z0-1]){5}/')]
+        #[With(pattern: '/([a-z0-1]){5}/')]
```

Comment:
I propose to use an extra PR for the renaming to `#[With]`

Reply:
Besides that, we can also add an internal map, so that you can only use minimum and maximum with an int for example + we can throw an exception if the properties are not typed, or do we already?

Reply:
so just move it first, rename it later you mean?

Reply:
Updated my comment, in case you missed it

Reply:
Yeah, should think of getting this smaller anyways, true. Refactoring wasn't as successful as hoped for